src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -69,9 +69,14 @@ internal OutstandingCallData(object taskCompletionSource, Action<JsonRpcMessage>
         private readonly Func<Task, object, JsonRpcMessage> handleInvocationTaskResultDelegate;
 
         /// <summary>
-        /// A map of the target object's method names to JsonRpcMethodAttribute.Name value.
+        /// A dictionary to map clr method names to <see cref="JsonRpcMethodAttribute" /> values, used to detect that same method shouldn't have multiple attribute values.
         /// </summary>
-        private readonly Dictionary<string, string> requestMethodToClrMethodMap;
+        private Dictionary<string, string> requestMethodToClrMethodMap = new Dictionary<string, string>(StringComparer.Ordinal);
+
+        /// <summary>
+        /// A dictionary to map <see cref="JsonRpcMethodAttribute" /> values to clr method names, used to detect that the same attribute value doesn't map to multiple methods.
+        /// </summary>
+        private Dictionary<string, string> clrMethodToRequestMethodMap = new Dictionary<string, string>(StringComparer.Ordinal);
```

Comment:
Also, we should probably use `ReadOnlyDictionary<string,string>` for these fields as a way to guarantee they don't change later on since we depend on them being thread-safe.